### DISCUSSION TITLE
Connector ideas

### DISCUSSION THREAD

**Discussion by Laskewitz** [microsoft/PowerPlatformConnectors#1014] [Ideas]
Do yo have an idea for a connector? 💡 What actions and triggers should the connector have? Let us know in the comments below! 👇

**Comment by kevmcdonk** [comment]
Canvas Apps connector for Azure Cosmos DB please!

**Reply by troystaylor** [reply]
Which endpoints are missing?: https://docs.microsoft.com/en-us/connectors/documentdb/#actions

**Reply by kevmcdonk** [reply]
Hmm, maybe things have changed and I have misunderstood the doc. At the top of that link, it has Power Apps listed as Not Available in the table of available places. When I tried previously, that was the case and you had to use Power Automate to call Cosmos DB and then call the flow from Power Apps.

**Reply by Laskewitz** [reply]
I found [this forum reply](https://powerusers.microsoft.com/t5/Building-Power-Apps/How-to-connect-POwerApps-with-COSMOSDB-connector/m-p/578984/highlight/true#M180695) that states it's not possible because Canvas Apps doesn't support unstructured data, so maybe that's the limitation. We won't be able to solve that with independent publisher connectors. 

@natalie-pienkowska, could you check if there are any plans to change support canvas apps with the Cosmos DB connector? Or is the limitation still blocking that?

**Reply by kevmcdonk** [reply]
Ah, that would make sense

**Reply by coatsy** [reply]
Could we revisit this now that we have `ParseJSON()`?

**Comment by lmheimendinger** [comment]
Retail system heartlandretail.com.  Need to access (RO) fields like store sales, credits, returns, etc.

**Reply by Laskewitz** [reply]
@lmheimendinger thanks for your suggestion. Will look into that one soon!

**Comment by ShanesCows** [comment]
How about a Pandora API for those of us that listen to cool tunes?

**Reply by troystaylor** [reply]
Certainly looks doable:
https://developer.pandora.com/docs/reference/graphql-api/overview/

**Reply by Laskewitz** [reply]
Great suggestion @ShanesCows. The "We're not accepting new access requests at this time. Please check again later." message at [this page](https://developer.pandora.com/partners/join/) might be a problem to get this soon though.

**Comment by CarstenGroth** [comment]
Supporting community to create more sustainability decision-making applications with low-code by adding connectors to Microsoft‘s Planetary Computer APIs 
https://www.microsoft.com/en-us/ai/ai-for-earth-tech-resources#primaryR3

**Reply by Laskewitz** [reply]
Very cool suggestion! Certainly will dive into that! Thanks, Carsten!

**Reply by troystaylor** [reply]
Hi Carsten,
The Species Classification API requires the content-type of application/octet-stream, which is only available in OpenAPI. Power Platform custom connectors currently have to be in the Swagger 2.0 format, so application/octet-stream is not available. 
Both the Camera Trap and Land Cover Mapping APIs are both only open to invited users, so I'm not able to build out those connectors.

**Comment by djouallah** [comment]
Google Sheet, at least for PowerQuery

**Reply by RebekkaAalbers** [reply]
He @djouallah ,
PowerQuery uses a different connector framework. For Power Apps and Power Automate there already is a Google Sheets connector:
https://docs.microsoft.com/en-us/connectors/googlesheet/

**Comment by mamurshe** [comment]
**Zoom**
A user [reported developing a custom connector ](https://powerusers.microsoft.com/t5/Connector-Development/Error-importing-zoom-openapi/m-p/1233914#M336)for it using the swagger provided by Zoom. I requested him to publish as an Independent Publisher, but he probably wasn't interested. This should be a quick one to do.

**Reply by troystaylor** [reply]
I can take a look at creating this one. Swagger is available at: https://marketplace.zoom.us/docs/api-reference/zoom-api/Zoom%20API.oas2.json

**Reply by troystaylor** [reply]
Provided file says Swagger 2.0, but includes oneOf's which are part of OpenAPI 3.0, among other issues.

**Comment by MMT0905** [comment]
GroupMe connector

It would be nice to have a GroupMe connector that will allow for posting of messages, and GET of messages posted.

**Reply by Laskewitz** [reply]
Thanks, great suggestion! I'm sure gonna take a look at this one. 😁

**Reply by MMT0905** [reply]
@Laskewitz any luck on this ? :)

**Reply by troystaylor** [reply]
If Daniel can't do it, I can add it to what I'm working on

**Reply by MMT0905** [reply]
oooh that would be good too!

**Reply by Laskewitz** [reply]
@MMT0905 Sorry - a bit of a late reply. I started on it, but am not quite finished yet. Hoping to finish it the upcoming weekend or the weekend after.

**Reply by MMT0905** [reply]
EEEk That's so exciting Thank you!

**Reply by troystaylor** [reply]
@Laskewitz how are you handling the implicit OAuth?

**Comment by Customery** [comment]
In my online course business, Customery Academy, I use Zapier to connect to the following apps:

Bonjoro - create task to record a personalised video to welcome new students
CloudConvert - turns a Google Slide into a PDF
ConvertKit - add a student as a new subscriber in my email marketing platform
New Zenler - my online course platform

Would be great to see connectors for more of the top email marketing (ConvertKit, Mailerlite, ActiveCampaign) and online course platforms (New Zenler, Teachable, Thinkific, Podia, Kartra, Kajabi).

**Reply by troystaylor** [reply]
Hi Neil,
There are several great services here that could be made, either manually or from a Swagger definition file, if available. For those Zapier-connected services, I don't see an open API for Bonjoro or New Zenler, so those probably aren't possible.

Troy

**Reply by natalie-pienkowska** [reply]
Thanks for sharing! Adding some of these into https://github.com/microsoft/PowerPlatformConnectors/wiki/Top-Connector-Asks , except for the ones that Troy mentioned are likely Zapier-specific.

**Reply by Customery** [reply]
I know Zenler has an API, it's [documented here](https://www.newzenler.com/api/documentation/public/api-doc.html). I'll ask the Bonjoro team to see if their API is public.

**Comment by sparkitect** [comment]
### ThoughtTrace
ThoughtTrace is a Document Understanding and Contract Analytics platform that uses pre-trained AI to read, organize, and manage the document workflows and processes critical to your business.

Triggers and actions for the Documents (and related) methods.

https://api.thoughttrace.com/swagger/index.html

**Reply by troystaylor** [reply]
I see the definition they provide is OpenAPI 3.0.1, so this will require a bit more work since Power Automate connectors use Swagger 2.0.

**Reply by troystaylor** [reply]
Can you please take a look at this file and use it to create a new custom connector?:
https://github.com/troystaylor/PowerPlatformConnectors/blob/ThoughtTrace/independent-publisher-connectors/ThoughtTrace/apiDefinition.swagger.json
I have signed up for a ThoughtTrace account, but as of now, I do not have a workspace I can test this connector against. Please keep in mind that you will need to configure the custom connector using OAuth 2.0, so you will need to obtain a client ID and secret. If it works for you, please post a screenshot with three successful actions. I will then work on cleaning up the Swagger file so that it is ready for submission to the Independent Publisher Connector program.

**Reply by sparkitect** [reply]
On it...
(between meetings)

**Reply by sparkitect** [reply]
Stuck here attempting to create a connection:

![image](https://user-images.githubusercontent.com/9206808/166008750-3f3be318-9c3c-4821-8ac2-4f8d2e1f2d2c.png)

(I'm new here, btw ;)

**Reply by troystaylor** [reply]
You will need to register the following as the redirect URL within ThoughtTrace OAuth settings:
https://global.consent.azure-apim.net/redirect

**Comment by schabungbam** [comment]
**Workday**
Got requests from customers on this one. The one we have internally is more complex than needed (it is what we call as 'codeful') connector.

**Comment by mcombp** [comment]
**Zoho CRM API V2.0** please
[https://www.zoho.com/crm/developer/docs/api/v2/modules-api.html](https://www.zoho.com/crm/developer/docs/api/v2/modules-api.html)
I've created this successfully using Postman Collection. This fails when importing the Postman Collection into a Custom Connector. The Request fails with the Response Body of: 

{  "code": "INVALID_REQUEST",
  "details": {},
  "message": "unable to process your request. please verify whether you have entered proper method name, parameter and parameter values.",
  "status": "error"}

**Comment by Laskewitz** [comment]
**IBM Watson**

IBM has a bunch of awesome cognitive services APIs. Would be cool to have them available in the Power Platform as well.

[IBM Watson Documentation](https://cloud.ibm.com/developer/watson/documentation)

**Reply by Laskewitz** [reply]
@troystaylor If you're looking for more APIs to fill your backlog... 😁

**Reply by troystaylor** [reply]
Always!

**Comment by martin-dreyer-teck** [comment]
A DataBricks query connector would be great, now that the SQL *Execution* REST API is in preview.

Actions should mirror these
- Execute Statement | POST /api/2.0/sql/statements/
- Get Status & Results | GET /api/2.0/sql/statements/{token}
- Cancel Execution | POST /api/2.0/sql/statements/{token}/cancel

**Comment by ChrisGarty** [comment]
Documentation for Top Connector Asks is here: https://github.com/microsoft/PowerPlatformConnectors/wiki/Top-Connector-Asks
Keep suggesting connectors to provide additions to that list!

**Comment by lmheimendinger** [comment]
Sorry not following your point.

**Comment by stevesint** [comment]
hello

power automate connector, telegram with "sendmediagroup" option, please!
